Sources/PopupBridge/POPPopupBridge.swift

```diff
@@ -0,0 +1,194 @@
+import Foundation
+import SafariServices
+import WebKit
+
+@objcMembers
+public class POPPopupBridge: NSObject, WKScriptMessageHandler, SFSafariViewControllerDelegate {
+    
+    // MARK: - Private Properties
+    
+    private let messageHandlerName = "POPPopupBridge"
+    private let hostName = "popupbridgev1"
+    
+    private let webView: WKWebView
+    private let delegate: POPPopupBridgeDelegate
+    private var safariViewController: SFSafariViewController?
+    
+    private static var scheme: String?
+    private static var returnBlock: ((URL) -> Bool)?
+    
+    // MARK: - Initializer
+    
+    // TODO: - make unfailable
+    public init?(webView: WKWebView, delegate: POPPopupBridgeDelegate) {
```

Comment:
👋 Yeh that's an interesting callout. I have[ this PR](https://github.com/braintree/popup-bridge-ios/pull/46) almost done for addressing the TODOs around requiring the `scheme` and for making the `init` non-failable.

IMO - I feel like I don't see inits that throw a ton in Apple's APIs, so they don't feel super natural to me. But we could do some research.

For this case, I don't think we need an init that throws since PopupBridge _must_ have a url scheme. Let me know if this makes sense!

Reply:
That makes sense - more just a general thought than anything else. You're right they don't use it much in their older frameworks (Foundation as an example) but they do see to use it some of their Swift first frameworks (examples: [file document SwiftUI init](https://developer.apple.com/documentation/swiftui/referencefiledocument/init(configuration:)) and [navigation path](https://developer.apple.com/documentation/swiftui/navigationpath/codablerepresentation/init(from:)) - looks like Airbnb uses it in some places as well like [here](https://github.com/airbnb/lottie-ios/blob/7484b4f4580116d7bf135b5b6097f38f801deaf8/Sources/Private/Model/ShapeItems/Shape.swift#L22) and [here](https://github.com/airbnb/lottie-ios/blob/e1a05adf37fb0557459a2628ef6089571d0bfbc0/Sources/Public/DotLottie/DotLottieFile.swift#L23)).

On the Apple side, it will be interesting to see if this is a pattern they follow or not when moving Foundation to Swift. Not a deal breaker either way, just an observation that we should explore if it makes sense for us at some point.